driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -19,46 +19,128 @@
+    /**
+     * The result of dividing {@code this} value by the {@code other} value.
+     * This is not integer division: dividing {@code 1} by {@code 2} will yield
+     * {@code 0.5}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression divide(NumberExpression other);
+
+    /**
+     * The result of dividing {@code this} value by the {@code other} value.
+     * This is not integer division: dividing {@code 1} by {@code 2} will yield
+     * {@code 0.5}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression divide(final Number other) {
+        return this.divide(Expressions.numberToExpression(other));
     }
 
-    NumberExpression add(NumberExpression n);
-
-    default NumberExpression add(final Number add) {
-        return this.add(Expressions.numberToExpression(add));
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression add(NumberExpression other);
+
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression add(final Number other) {
+        return this.add(Expressions.numberToExpression(other));
     }
 
-    NumberExpression subtract(NumberExpression n);
-
-    default NumberExpression subtract(final Number subtract) {
-        return this.subtract(Expressions.numberToExpression(subtract));
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression subtract(NumberExpression other);
+
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default NumberExpression subtract(final Number other) {
+        return this.subtract(Expressions.numberToExpression(other));
     }
 
-    NumberExpression max(NumberExpression n);
-
-    NumberExpression min(NumberExpression n);
-
+    /**
+     * The larger value of {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression max(NumberExpression other);
+
+    /**
+     * The smaller value of {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    NumberExpression min(NumberExpression other);
+
+    /**
+     * The integer result of rounding {@code this} to the nearest even value.
+     *
+     * @return the resulting value.
+     */
     IntegerExpression round();
 
+    /**
+     * The result of rounding {@code this} to the nearest even {@code place}.
```

Comment:
This reads like we are rounding to "the nearest even place", while actually we are rounding to the specified "decimal place" using the "round half to even" approach.

Reply:
I am not sure what the wording should be here, and in the overload. Java uses the wording:

> Rounding mode to round towards the "nearest neighbor" unless both neighbors are equidistant, in which case, round towards the even neighbor. 

So perhaps

```
The result of rounding {@code this} using the "half even" rounding mode, 
which rounds towards the "nearest neighbor" unless both neighbors are 
equidistant, in which case, it rounds towards the even neighbor.
```

Reply:
[marked optional]

_"The result of rounding `{@code this}` to `{@code place}` decimal places using the "half to even" approach."_ addresses the issue of _"nearest even"_ appearing to be applied to _"`{@code place}`"_ in _"nearest even `{@code place}`"_, which I talked about in the first comment.

Reply:
Used this for `round(IntegerExpression place)`, but not for `round()` (if that should be updated too, let me know).